src/PhpSpreadsheet/Settings.php

```diff
@@ -166,23 +161,19 @@ public static function getCache()
         return self::$cache;
     }
 
-    /**
-     * Set the HTTP client implementation to be used for network request.
-     */
-    public static function setHttpClient(ClientInterface $httpClient): void
```

Comment:
Removed this because it must be provided at runtime by the user. Some HTTP clients implementing PSR-18 might be scoped, so it's safer to inject it at runtime instead of keeping a client in the memory

Reply:
> might be scoped

What to you mean be that ? And what difference does it make for PhpSpreadsheet if an HTTP client is scoped ?

Reply:
When using Symfony HTTP Client, it can be scoped, meaning it can have a base uri, default headers, etc., which might not be suitable for PhpSpreadsheet usage. Some HTTP clients can also contain state: history, cookies, etc., so keeping its state in the memory can have side-effects, such as leaking state, etc.